main/src/main/java/cgeo/geocaching/unifiedmap/UnifiedMapActivity.java

```diff
@@ -646,13 +646,15 @@ public static void refreshWaypoints(final UnifiedMapViewModel viewModel) {
     }
 
 
-    public void addSearchResultByGeocaches(final SearchResult searchResult) {
-        Log.d("add " + searchResult.getGeocodes());
+    public void replaceSearchResultByGeocaches(final SearchResult searchResult) {
+        Log.d("replace " + searchResult.getGeocodes());
+        viewModel.caches.write(true, caches -> { // use post to make it background capable
+            caches.clear();
```

Comment:
I use caches.clear here. Set::clear didn't work for me - no idea why, I will check again...